src/index.js

```diff
@@ -1,234 +1,258 @@
+const checkForUpdates = function () {
+    logger.log('Checking for updated repositories');
     Cache.getCache().then((cache) => {
-        let repoChecks = [];
-        for(let job of notifyServices) {
-            repoChecks.push(checkRepository(job, cache[job.image.user + "/" + job.image.name]));
+        const repoChecks = [];
+        for (const job of notifyServices) {
+            let key = job.image.user + '/' + job.image.name;
+            if (job.image.tag) {
+                key += ':' + job.image.tag;
+            }
+            logger.log('Checking: ', key);
+            repoChecks.push(checkRepository(job, cache[key]));
         }
         Promise.all(repoChecks).then((checkResult) => {
-            let newCache = {};
-            let updatedRepos = [];
-            for(let res of checkResult) {
-                newCache[res.user + "/" + res.name] = {
+            const newCache = {};
+            const updatedRepos = [];
+            for (const res of checkResult) {
+                let key = res.user + '/' + res.name;
+                const cacheObj = {
                     user: res.user,
                     name: res.name,
                     lastUpdated: res.lastUpdated
                 };
-                if(res.updated) {
-                    let updatedString = res.user == "library" ? res.name : res.user + "/" + res.name;
+
+                if (res.tag) {
+                    key += ':' + res.tag;
+                    cacheObj.tag = res.tag;
+                }
+
+                newCache[key] = cacheObj;
+
+                if (res.updated) {
+                    let updatedString = res.user == 'library' ? res.name : res.user + '/' + res.name;
+                    if (res.tag) {
+                        updatedString += ':' + res.tag;
+                    }
                     updatedRepos.push({
                         job: res.job,
                         updatedString: updatedString
                     });
                 }
             }
             Cache.writeCache(JSON.stringify(newCache)).then(() => {
-                if(updatedRepos.length > 0) {
-                    updatedRepos.forEach(o => o.job.actions.forEach(o2 => {
-                        if(o2.type == "webHook"){
-                            let webHook = config.webHooks[o2.instance];
-                            let options = {method: webHook.httpMethod, uri: webHook.reqUrl};
+                if (updatedRepos.length > 0) {
+                    updatedRepos.forEach((o) => o.job.actions.forEach((o2) => {
+                        if (o2.type == 'webHook'){
+                            const webHook = config.webHooks[o2.instance];
+                            const message = webHook.httpBody;
+                            Object.keys(message).forEach((key) => {
+                                message[key] = message[key].replace('$msg', 'Docker image \'' + o.updatedString + '\' was updated:\n' + JSON.stringify(o.job.image));
```

Comment:
Unfortunately, I just found a bug here. In case you have a key/value pair in your HTTP Body configured in your config.json where a key is not a string, this will cause an error. Take for example this payload:
`"httpBody": {
                "foo": {
                    "a": true
                }
            }`

So we need to change it so that we only replace it in places where the value is a string.

Reply:
Fixed